packages/block-library/src/social-links/edit.js

```diff
@@ -38,24 +39,48 @@ const sizeOptions = [
 	{ name: __( 'Huge' ), value: 'has-huge-icon-size' },
 ];
 
+const getDefaultBlockLayout = ( blockTypeOrName ) => {
+	const layoutBlockSupportConfig = getBlockSupport(
+		blockTypeOrName,
+		'__experimentalLayout'
+	);
+
+	return layoutBlockSupportConfig?.default;
+};
+
 export function SocialLinksEdit( props ) {
 	const {
 		attributes,
 		iconBackgroundColor,
 		iconColor,
 		isSelected,
 		setAttributes,
+		name,
 		setIconBackgroundColor,
 		setIconColor,
+		clientId,
 	} = props;
 
 	const {
 		iconBackgroundColorValue,
 		iconColorValue,
-		itemsJustification,
 		openInNewTab,
 		size,
+		layout,
 	} = attributes;
+	// This probably should not be used/checked here. We need to make all `social-links`
+	// blocks to use `flex` layout.
+	const { themeSupportsLayout } = useSelect(
+		( select ) => {
+			const { getSettings } = select( blockEditorStore );
+			return {
+				themeSupportsLayout: getSettings()?.supportsLayout,
+			};
+		},
+		[ clientId ]
+	);
+
+	const usedLayout = !! layout || getDefaultBlockLayout( name );
```

Comment:
All this logic is something that I wanted to do automatically in the hook for existing blocks using "layout", the problem is that we don't have a filter for `InnerBlocks` to do it. Maybe we should introduce an unstable filter to be able to inject the layout prop into InnerBlocks.